torch/csrc/api/src/module.cpp

```diff
@@ -0,0 +1,81 @@
+#include <torch/nn/module.h>
+#include <torch/nn/cursor.h>
+
+#include <torch/csrc/autograd/variable.h>
+
+#include <ATen/ATen.h>
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace torch { namespace nn {
+
+Module::Module(std::string name)
+    : name_(std::move(name)), is_training_(false) {}
+
+Module::~Module() = default;
+
+std::unique_ptr<Module> Module::clone() {
+  AT_ERROR("clone() has no default implementation");
+}
+
+// Train/Eval mode
+void Module::train() {
+  is_training_ = true;
+}
+
+void Module::eval() {
+  is_training_ = false;
```

Comment:
Is there really any reason to avoid putting such trivial methods in the header? They really should be possible to inline

Reply:
I would be surprised if methods that have a single assignment inside increased the compile time significantly. Ok, we can just put everything in `.cpp` for now. I guess it's not a performance critical part.